core/server/kustomization.go

```diff
@@ -11,33 +11,38 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type defaultClusterNotFound struct{}
+
+func (e defaultClusterNotFound) Error() string {
+	return "default cluster not found"
+}
+
 func (cs *coreServer) ListKustomizations(ctx context.Context, msg *pb.ListKustomizationsRequest) (*pb.ListKustomizationsResponse, error) {
 	clustersClient := clustersmngr.ClientFromCtx(ctx)
 
-	clist := clustersmngr.NewClusteredList(func() client.ObjectList {
-		return &kustomizev1.KustomizationList{}
-	})
+	if msg.Namespace != "" {
+		res, err := listKustomizationsInNamespace(ctx, clustersClient, msg.Namespace)
 
-	if err := clustersClient.ClusteredList(ctx, clist, client.InNamespace(msg.Namespace)); err != nil {
-		return nil, err
+		return &pb.ListKustomizationsResponse{
+			Kustomizations: res,
+		}, err
 	}
 
 	var results []*pb.Kustomization
 
-	for n, l := range clist.Lists() {
-		list, ok := l.(*kustomizev1.KustomizationList)
-		if !ok {
-			continue
-		}
+	if _, found := cs.cacheContainer.Namespaces()[clustersmngr.DefaultCluster]; !found {
+		return nil, defaultClusterNotFound{}
+	}
 
-		for _, kustomization := range list.Items {
-			k, err := types.KustomizationToProto(&kustomization, n)
-			if err != nil {
-				return nil, fmt.Errorf("converting items: %w", err)
-			}
+	for _, ns := range cs.cacheContainer.Namespaces()[clustersmngr.DefaultCluster] {
```

Comment:
do we need to call `cs.cacheContainer.Namespaces()` twice?

can we make line 33:
```
namespaces, found := cs.cacheContainer.Namespaces()[clustersmngr.DefaultCluster]
if !found {
```

and then just `for _, ns := range namespaces {`

Reply:
We can, but it does not have much toll on it, it's a "static" map. Will update.

Reply:
lol yes i was not worried about the poor code getting tired 🤣